### PR TITLE
Test with dmd-beta and dmd-2.087.1

### PR DESCRIPTION
**Pull request by Geod24**
DMD 2.087.1 should solve all the C++ regressions that we had.
    In addition, testing with dmd-beta should allow us to catch regressions early.
    It will slow down our CI, however if dmd-beta introduce a regression it won't fail the build.
    We do look at the logs often enough that we'd catch any issue.

### CODE REVIEW

**Comment by Geod24** [comment]
This uncover a DMD regression. Fix on its way at: https://github.com/dlang/dmd/pull/10256
Let's wait until the beta actually passes to merge this.

**Comment by Geod24** [comment]
And also https://github.com/dlang/dmd/pull/10257 because I'm a dingus.

**Comment by Geod24** [comment]
DMD v2.087.1 has been released, so I added it to this patch.

**Comment by Geod24** [comment]
💚

**Review by AndrejMitrovic — APPROVED** [review]

**Comment by Geod24** [comment]
`dmd-beta` is red because it's the old beta (without my fixes).
But `2.087.1` is newer

**Pull request merged**
